### ISSUE TITLE
[FEATURE] Add a note to responses

### ISSUE THREAD

**Issue by jobenjada** [formbricks/formbricks#332]
### Overview
Users requested to add notes to singles submissions so that the team can briefly discuss or document how to react to feedback.

### Detailed Description
Currently, the "Responses" View looks like this:
![grafik](https://github.com/formbricks/formbricks/assets/72809645/76581be2-8298-424a-bb8b-934ce7f6888e)

We'd like to offer the possibility to add notes to each submission. Here is a mock up of what the submission should look like **on hover**. The card slides out from left to right with a CSS animation:
![grafik](https://github.com/formbricks/formbricks/assets/72809645/55deb5b5-54a5-4b16-8ce1-10414db89d08)

On click on the card, the note window slides out and scales up to fit next to the submission:
![grafik](https://github.com/formbricks/formbricks/assets/72809645/39f89dcf-5db6-4df4-8f72-8f4b8fd43a78)

The note window behaves as follows:

1. A click into the window catches the focus and the user can type something. On ENTER, the message is submitted and saved. Once its saved,  the meta data is added (name and time of writing). As of now, notes cannot be edited.
2. A click on the crossed eye icon in the top right corner hides the card. Ideally, the animation goes backwords (shrink and slide under the response card).

When a submission has a note, the state looks like this:
![grafik](https://github.com/formbricks/formbricks/assets/72809645/b7f01536-3913-466a-bf77-29083054e71e)

On hover, it slides out a few pixels:
![grafik](https://github.com/formbricks/formbricks/assets/72809645/9291e0be-2be9-4637-a473-468b7ed82593)

On click, it opens in the same way a new note openes (with slide out and grow animation).

### Tasks
1. Add a model "ResponseNote" to the Prisma Model with the fields `id`,  `createdAt`,  `updatedAt`,  `response`,  `responseId` , `userId`, `user`, `text` and link it to the Response & User models
3.  Create the UI mocked up above
4. Test

### Setting up dev environment
Here are instructions on how to start the dev environment: 

https://formbricks.com/docs/contributing/setup

**Comment by gitstart-formbricks** [comment]
Hello @jobenjada. Is there a link to the design to be able to get specific details like color codes, etc? Or we rely on the screenshots. Thanks.

**Comment by jobenjada** [comment]
We're using Tailwind Colors. The gray scale is the `slate` color group, the yellow is amber-50. The Formbricks brand color is "brand-dark".

Thanks!

**Comment by gitstart-formbricks** [comment]
Since the visible notes card has the eye icon crossed through which implies hide the card, We have been thinking of adding the eye icon(not crossed through) on the notes card on hover to imply view card.

**Comment by jobenjada** [comment]
Overall: Good work! We're almost there :) Here is the feedback:

1. Response Width
In the mockup you see that the response card is fullwidth, before the note appears. In your draft, the card is shortened by default to make room for the note. Please adjust the width of the card dynamically, when then note card is shown. However, its okay to shorten it a bit by default to make room for the note indicator. In other words:

 - Shorten card by a few % to make room for the indicator which pops out hover
 - While the Note Card pops out, shorten the Response Card to make room for the Note Card
 - When the Note Card disappears, increase width of Response Card back to default.
 
![grafik](https://github.com/formbricks/formbricks/assets/72809645/04d90770-7920-4344-9d82-6de6ae4b2e4b)

2. Appearing Animation: On hover the note indicator appears (and disappears) without a fade. Please fade it in and out.

3. Center Note indicator vertically
![grafik](https://github.com/formbricks/formbricks/assets/72809645/b7330ef1-a8af-4e30-b0b0-2f21ce6b7dbc)

4. Please change the cursor to pointer when hovering the note indicator

5. Reduce icon size of Eye to the size of the Trash Can Icon h-4 w-4

6. Prevent the form from submitting when field is empty
![grafik](https://github.com/formbricks/formbricks/assets/72809645/b0875568-af38-4675-9755-1ca43b1f7d81)

7. Add "text-sm" to "Note"
![grafik](https://github.com/formbricks/formbricks/assets/72809645/64fc2eb6-b037-409e-9ede-b412868986d0)

The rest looks great! Keep up the good work 🚀

**Comment by jobenjada** [comment]
Hey Gitstart team!

Thanks for the PR, we're almost done 👍  One last thing:

In the ticket I wrote that the default state of the response card has to indicate that a note was added, if one was added:
![grafik](https://github.com/formbricks/formbricks/assets/72809645/0519f789-98d2-4aa3-90a9-948b1615cda5)

The current default state with added note looks like this (and is indistinguishable from a response without a note)
![grafik](https://github.com/formbricks/formbricks/assets/72809645/6f54d757-a9e3-4020-9b24-adf9f8e71a21)

The reason behind it is that it's impossible for a different team member to know if a question has a note ore not. You'd have to hover over all responses just to find out if someone in your team left a note or not.

Pls update that - the rest looks great - good work!! 🤸
